reacnetgenerator/_detect.py

```
@@ -400,6 +407,236 @@ def _getbondfromase(self, step_atoms: Atoms, cell: np.ndarray):
         return bond, bondlevel
 
+    def _new_openbabel_molecule(
+        self,
+        step_atoms: Atoms,
+        cell: np.ndarray,
+    ):
+        """Create an Open Babel molecule while preserving input atom IDs."""
+        mol = openbabel.OBMol()
+        mol.BeginModify()
+        for idx, (num, position) in enumerate(
+            zip(step_atoms.get_atomic_numbers(), step_atoms.positions)
```

Ruff (0.16.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)